Common++/header/MacAddress.h

```diff
@@ -5,7 +5,7 @@
 #include <string.h>
 #include <string>
 
-#if __cplusplus > 199711L
+#if __cplusplus > 199711L || _MSC_VER >= 1800
```

Comment:
so just to make sure: you added `_MSC_VER` because it's more reliable than using `__cplusplus`?

In that case, I'm not sure why `std::copy` fails

Reply:
Yes __cplusplus reports 199711L on any version of VS up to 2017.3. Even in that version it lies unless you add a special compiler flag see this article:

https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/

The std::copy does not FAIL. I mentioned this on our other thread before but im copying it here:

When i flipped that switch it updated your code to use some newer C++ features specifically some code using std::copy. Apparently microsoft flagged this particular usage of std::copy as potentially risky because it can't check bounds in debug mode. See article discussing this kind of issue here https://devblogs.microsoft.com/cppblog/why-am-i-getting-these-_scl_secure_no_warnings-messages/ . You can either change your call to std::copy, or disable SDL, or similar . I will leave the remaining googling and decision to you.

Reply:
sure, I'll build the code and see what's the best option